static/js/widgets/GreetingsWidget.js

```diff
@@ -0,0 +1,67 @@
+
+define([
+		"dojo/_base/declare",
+		"dojo/_base/lang",
+		"dojo/_base/array",
+		"dojo/on",
+		"dojo/request/xhr",
+		"dojo/cookie",
+		"dijit/_WidgetBase",
+		"dijit/_TemplatedMixin",
+		"dojo/text!./templates/GreetingsWidget.html"
+	], function(declare, lang, arrayUtil, on, xhr, cookie, _WidgetBase, _TemplatedMixin, template){
+	return declare("GreetingWidget", [_WidgetBase, _TemplatedMixin],{
+		greetingName: 'No name',
+		author: "An anonymous person",
+		deleteUrl: require.toUrl("../static/img/delete.png"),
+		updateUrl: require.toUrl("../static/img/update.png"),
+		templateString: template,
+		baseClass: "greetingWidget",
+
+		postCreate: function(){
+			this.inherited(arguments);
+			this.own(
+				on(this.deleteUrlNode, 'click', lang.hitch(this, "_deleteGreeting")),
+				on(this.updateUrlNode, 'click', lang.hitch(this, "_updateGreeting"))
+			)
+		},
+
+		_setDeleteUrlAttr: function(imagePath){
+			if (imagePath != ''){
+				this._set("deleteUrl", imagePath);
+				this.deleteUrlNode.src = imagePath;
+			}
+		},
+
+		_setUpdateUrlAttr: function(imagePath){
+			if (imagePath != ''){
+				this._set("updateUrl", imagePath);
+				this.updateUrlNode.src = imagePath;
+			}
+		},
+
+		_showGreetingDetail: function(){
+
```

Comment:
nếu là abstract thì thêm chú thích : `//abstract method`, nếu là hàm để chỗ khác listen event, thì cũng chú thích sẽ được listen ở đâu cho dễ control, chưa implement thì : `//TODO implement this function `